src/pypnusershub/auth/authentication.py

```diff
@@ -0,0 +1,196 @@
+    """
+
+    @property
+    def id_provider(self) -> str:
+        """
+        Identifier of the authentication provider.
+
+        Returns
+        -------
+        str
+            The authentication provider identifier.
+
+        Raises
+        ------
+        NotImplementedError
+            This method must be implemented by subclasses.
+        """
+        raise NotImplementedError()
+
+    @property
+    def label(self) -> str:
+        """
+        Label of the authentication provider.
+
+        Returns
+        -------
+        str
+            The label of the authentication provider.
+        """
+
+    @property
+    def login_url(self) -> str:
+        """
+        External logout URL.
+        Must be define if the authentication provider is external, otherwise put an empty string
+        Raises
+        ------
+        NotImplementedError
+            This method must be implemented by subclasses.
+        """
+        raise NotImplementedError()
+
+    @property
+    def logout_url(self) -> str:
+        """
+        External logout URL.
+        Must be define if the authentication provider is external, otherwise put an empty string
+
+        Raises
+        ------
+        NotImplementedError
+            This method must be implemented by subclasses.
+        """
+        raise NotImplementedError()
+
+    @property
+    def logo(self) -> str:
+        """
+        Logo of the authentication provider.
+
+        Returns
+        -------
+        str
+            URL of the logo image.
+        """
+        return ""
+
+    @property
+    def is_uh(self) -> bool:
+        """
+        Return whether the authentication is an 'usershub-auth-module' authentication.
+
+        Returns
+        -------
+        bool
+        """
+        return True
+
+    def authenticate(self, *args, **kwargs) -> models.User:
+        """
+        Authenticate a user with the provided parameters.
+
+        Parameters
+        ----------
+        *args : Any
+            Positional arguments to be passed to the implementation.
+        **kwargs : Any
+            Keyword arguments to be passed to the implementation.
+
+        Raises
+        ------
+        NotImplementedError
+            This method must be implemented by subclasses.
+
+        Returns
+        -------
+        models.User
+            The result of the authentication process, which must be a User object.
+        """
+        raise NotImplementedError()
+
+    def revoke(self) -> Any:
+        """
+        Revoke current authentication.
+
+        Raises
+        ------
+        NotImplementedError
+            This method must be implemented by subclasses.
+
+        Returns
+        -------
+        Any
+            Revocation result depending on the implementation.
+        """
+        raise NotImplementedError()
+
+    def authorize(self) -> Any:
+        raise NotImplementedError()
+
+
+class DefaultConfiguration(Authentication):
```

Comment:
```suggestion
class InternalAuthentication(Authentication):
```
Ou alors `DatabaseAuthentication` ?

Reply:
InternalAuthentication me semble plus clair :+1: